src/xkb_bell.c

```diff
@@ -76,8 +79,10 @@ void initialize_xkb(void) {
 		return;
 	}
 
-	if (! XkbOpenDisplay(DisplayString(dpy), &xkb_base_event_type, &ir,
-	    NULL, NULL, &reason) ) {
+	xkb_disp = XkbOpenDisplay(DisplayString(dpy), &xkb_base_event_type, &ir,
+		NULL, NULL, &reason);
+
+	if (!xkb_disp) {
```

Comment:
Oh. Oooooooooh. So keeping the connection open is not relevant for xkeyboard to work. That certainly makes things a bit easier :)